src/enclave_proc_comm.rs

```diff
@@ -157,32 +158,25 @@ where
         .map_err(|e| {
             e.add_subaction("Failed to send command to all enclave processes".to_string())
         })?
-        .iter_mut()
-        .map(|socket| {
+        .into_iter()
+        .map(|mut socket| {
+            // Send the command.
+            enclave_proc_command_send_single(cmd, args, socket.borrow_mut())?;
```

Comment:
It's more logical if the code is structured as: epoll ctl add, followed by enclave_proc_command_send_single, so that the thing that waits for events is created before the thing that generates the events

Reply:
I agree, but maintaining the logical order clutters the code. Any interaction with epoll is done with a RawFd, so we need to be extra careful to not close the socket (done automatically when the Socket object is dropped) and invalidate its fd. To keep the order you mentioned, I see three viable options:
- copy the socket and feed the epoll with the clone (this version is currently in mainline)
- have the map return socket's fd (and consume the socket): each closure ends with Ok(socket.into_raw_fd())
  - definitely the best solution, but some extra comments are needed for this line to explain why this is necessary + the result is never actually used / needed
- have the map return all sockets: each closure ends with Ok(socket). In this case, we also need to add some extra logic to guarantee that these sockets are used when we wait on epoll

I honestly prefer the version written in this commit, but if you believe otherwise, I can implement one of these options